doc/sphinx/src/building.rst

```diff
@@ -3,3 +3,216 @@
++------------------------------------------+------------+---------+                                               |
+| SINGULARITY_KOKKOSKERNERNELS_INSTALL_DIR | string     | NONE    |                                               |
++------------------------------------------+------------+---------+-----------------------------------------------+
+| SINGULARITY_KOKKOSKERNELS_SUB_DIR        | string     | NONE    | Set this to build kokkos-kernels "in-tree"    |
+|                                          |            |         | by adding it as a subdirectory.               |
++------------------------------------------+------------+---------+-----------------------------------------------+
+
+Dependencies
+------------
+
+``singularity-eos`` has a number of dependencies that are handled in a
+number of different ways:
+* `spiner`_ is a required dependency, included as a submodule
+* `hdf5`_ is an optional dependency. It is needed for the table
+  readers. If you want it, it must be installed externally and
+  findable by ``cmake``. ``MPI`` is an optional dependency of
+  ``hdf5``, but otherwise not needed.
+* `eospac`_ is an optional dependency. This is needed if you want to
+  use sesame tables. If you want it, it must be installed externally
+  and findable by ``cmake``
+* `kokkos`_ is an optional dependency. It provides GPU support. If it's
+  available externally, ``singularity-eos`` will use the available
+  version. If not, ``singularity-eos`` will use its own version,
+  packaged as a submodule.
+* `Eigen`_ is an optional dependency and is used for linear algebra on
+  the CPU when doing mixed-cell closures. If it's available
+  externally, ``singularity-eos`` will use the available version. If
+  not, ``singularity-eos`` will use its own version, packaged as a
+  submodule.
+* `kokkos-kernels`_ is an optional dependency. This must be available
+  externally if desired, but there are a number of ways to expose
+  it. One can set ``SINGULARITY_KOKKOSKERNELS_SUB_DIR`` to tell
+  ``cmake`` where to ``add_subdirectory`` to make it available. One
+  can also simply let ``cmake`` find a pre-installed version of the
+  library.
+* A fortran compiler is required if fortran bindings are enabled.
+
+.. _spiner: https://github.com/lanl/spiner
+
+.. _hdf5: https://www.hdfgroup.org/solutions/hdf5/
+
+.. _eospac: https://laws.lanl.gov/projects/data/eos/eospacReleases.php
+
+.. _kokkos: https://github.com/kokkos/kokkos
+
+.. _Eigen: https://eigen.tuxfamily.org/index.php?title=Main_Page
+
+.. _kokkos-kernels: https://github.com/kokkos/kokkos-kernels/
+
+If you use spack, but would like to build ``singularity-eos`` from
+source, you can install dependencies via, e.g.,
+
+.. code-block:: bash
+  git clone --recursive git@github.com:lanl/singularity-eos.git
+  spack repo add singularity-eos/spack-repo
+  spack install --only dependencies singularity-eos+cuda cuda_arch=70
+
```

Comment:
If I have a checkout of singularity-eos and I want to build in a non-spack build directory, is there still a way to get the cmake options to match the spack options @mauneyc-LANL ? When there was a deps package, there was a cmake configuration file generated to aid that process, does such a thing still exist, or can we specify via spack where we want the whole singularity-eos spackage to be built?

Reply:
This is a good question that I would also like to know.

Reply:
spack has a `install_tree` config that we could modify on a user level, but I don't know if that's the best approach you guys want here

Reply:
This feature is not included in the `package.py`, tho I can't recall why I cut it. 

As for controlling where it's build, that would require a global configuration change. Something like

```bash
$ spack location --install-dir gcc
~/spack/opt/spack/linux-centos7-x86_64/gcc-4.9.3-iy4rw...
```

would be helpful in the short term. Long-term, probably something like what `-deps` was doing seemed useful for devs, so we can return to it

Reply:
could we just have both, @mauneyc-LANL ? A deps package and a full package? And the devs can just install the deps package?

Reply:
We could but I don't think it's necessary. Let's discuss some options later today (if you're around). I can try to mock up something later today.

Reply:
yep sounds good

Reply:
there is the option of having `singularity-eos-deps` spackage and inherit from it to make the `singularity-eos` spackage... at least you won't have to maintain in two places

Reply:
Right that's what I was thinking.

Reply:
Definitely don't want to maintain 2 packages, but I'd like to hear what @mauneyc-LANL is thinking. I know the spack people want folks to move away from `-deps` spackages and I'm assuming they have reasons and I'd like to understand that better before going against the grain.